pkg/analysis_server/test/services/correction/fix_test.dart

```diff
@@ -5943,6 +5943,44 @@ main() {
 ''');
   }
 
+  test_removeEmptyStatement_insideBlock() async {
+    String src = '''
+void foo() {
+  while(true) {
+    /*LINT*/;
+  }
+}
+''';
+    await findLint(src, LintNames.empty_statements);
+
+    await applyFix(DartFixKind.REMOVE_EMPTY_STATEMENT);
+
+    verifyResult('''
+void foo() {
+  while(true) {
+  }
+}
+''');
+  }
+
+  test_removeEmptyStatement_outOfBlock() async {
```

Comment:
Just deleting the empty statement here changes the semantics of the code, which is something we try to avoid. Might be better to replace `;` with `{}` when the parent is not a block. What do you think?